File: src/test/scala/algo/stack/FixedArrayStackTest.scala
package algo.stack

import org.scalatest.FunSuite

/**
 * Test for {@link FixedArrayStack}
 *
 * @author akhalikov
 */
class FixedArrayStackTest extends FunSuite {
  test("push and pop basic test") {
    val stack = new FixedArrayStack[String](3)
    stack.push("one")
    stack.push("two")
    assert("two".equals(stack.peek()))
    assert(stack.isEmpty == false)
    assert(stack.push("three") == true)
    assert("three".equals(stack.peek()))
    assert("three".equals(stack.pop()))
    assert("two".equals(stack.pop()))
    assert("one".equals(stack.pop()))
    assert(stack.isEmpty)
  }
}

File: src/test/java/algo/stack/FixedArrayStackTest.java
package algo.stack;

import org.junit.Test;

import static org.junit.Assert.*;

public class FixedArrayStackTest {

  @Test
  public void test1() throws Exception {
    FixedArrayStack<Integer> stack = new FixedArrayStack(3);
    assertTrue(stack.isEmpty());
    stack.push(1);
    stack.push(2);
    stack.push(3);
    assertFalse(stack.isEmpty());
    assertEquals(3, (int) stack.pop());
    assertEquals(2, (int) stack.pop());
    assertEquals(1, stack.size());
    assertFalse(stack.isEmpty());
    assertEquals(1, (int) stack.peek());
    assertEquals(1, (int) stack.pop());
    assertTrue(stack.isEmpty());
  }
}